### ISSUE TITLE
Proposal: Disable 'Pipe chain should start with raw value' in some situations

### ISSUE THREAD

**Issue by dylan-chong** [rrrene/credo#488]
Basically, the 'Pipe chain should start with the raw value' check produces undesired results

Example:

```
     Stream.repeatedly(fn -> char end) # Pipe chain should ...
      |> Enum.take(n)
      |> to_string()
```

I prefer the above code style versus

```
      fn -> char end
      |> Stream.repeatedly()
      |> Enum.take(n)
      |> to_string()
```

Or another example where i prefer the code where credo makes the warning

```
new(a: 1, b: 2) # Pipe chain should ...
|> f()
|> g()
```

```
[a: 1, b: 2] # This looks really strange
|> new()
|> f()
|> g()
```

See issue #280  for more context/more examples

# Potential solution 

If the argument creates a new `fn` or keyword list, do not show the warning.

If the argument is a variable, continue with the existing behaviour

**Comment by dylan-chong** [comment]
@rrrene

**Comment by rrrene** [comment]
@dylan-chong 
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

See the new config setting in `[.credo.exs](https://github.com/rrrene/credo/blob/3535143bc6b594eb00f8d592b0a60db847cb9bef/.credo.exs#L115)` to see how it works!

**Comment by dylan-chong** [comment]
Oops, probably shouldn't have closed until you released?
